### DISCUSSION TITLE
Using read_csv_auto with memory data

### DISCUSSION THREAD

**Discussion by ricardoekm** [duckdb/duckdb#7339] [Q&A] [answered]
Is it possible to use read_auto (e.g. read_json_auto, read_csv_auto) in Java using memory data? I'm loading data from S3, this would avoid a intermediate step of creating an temporary file in disk and asking DuckDB to load back into memory again.

It relates to https://github.com/duckdb/duckdb/pull/2844

Thanks!

**Comment by Mause** [comment, marked as answer]
Have you looked into using the httpfs extension to read directly from s3 within DuckDB?

**Reply by ricardoekm** [reply]
The httpfs sounds interesting! Thanks for the reply. It seems at this stage it support only CSV and Parquet, but not JSON? Is there something similar for JSON?

**Reply by Mause** [reply]
It supports any file type DuckDB can read - are you seeing something in the docs that implies otherwise?

**Reply by ricardoekm** [reply]
_With the httpfs extension, it is possible to directly query files over HTTP(S). This currently works for CSV and Parquet files._

https://duckdb.org/docs/extensions/httpfs.html
